docs/databases/swift-kuery.html

```diff
@@ -252,237 +229,150 @@ <h2 class="heading-2"><span class="blue-text">Step 5 (Optional):</span> Test sav
-
-  }</code></pre>
-      <p>Once again our project will now compile but nothing will happen as we haven't added any logic to our function.</p>
-      <p>Just like we did in step 4 we need to get a connection from the connection pool:</p>
-      <pre><code class="language-swift">func getTitles(_ callback: @escaping (String) -> Void ) -> Void {
-    pool.getConnection() { connection, error in
-      guard let connection = connection else {
-        guard let error = error else {
-          Log.error("Error connecting: Unknown Error")
-          return
-        }
-        Log.error("Error connecting: \(error)")
-        return
-      }
-    }
-  }</code></pre>
-      <p>Now we can build our query that will query the database for every entry in the "Titles" database:</p>
-      <pre><code class="language-swift">func getTitles(_ callback: @escaping (String) -> Void ) -> Void {
-    pool.getConnection() { connection, error in
-      guard let connection = connection else {
-        guard let error = error else {
-          Log.error("Error connecting: Unknown Error")
-          return
-        }
-        Log.error("Error connecting: \(error)")
-        return
-      }
-      let selectQuery = Select(from: titles)
-    }
-  }</code></pre>
+    // Write query and execute it here
+}</code></pre>
+      <p>Now we can build our SELECT query that will query the database for every entry in the "BookTable":</p>
+      <pre><code class="language-swift">let selectQuery = Select(from: App.bookTable)</code></pre>
       <p>Like before we can now execute the query using the `execute` method in Connection:</p>
-      <pre><code class="language-swift">func getTitles(_ callback: @escaping (String) -> Void ) -> Void {
-    pool.getConnection() { connection, error in
-      guard let connection = connection else {
-        guard let error = error else {
-          Log.error("Error connecting: Unknown Error")
-          return
-        }
-        Log.error("Error connecting: \(error)")
-        return
-      }
-
-      let selectQuery = Select(from: titles)
-
-      connection.execute(query: selectQuery) { selectResult in
-        guard let resultSet = selectResult.asResultSet else {
-          guard let error = selectResult.asError else {
-            return callback("Error executing query: Unknown Error")
-          }
-          return callback("Error executing query: \(error)")
-        }
-      }
+      <pre><code class="language-swift">connection.execute(query: selectQuery) { selectResult in
+    guard let resultSet = selectResult.asResultSet else {
+        Log.error("Error connecting: \(selectResult.asError?.localizedDescription ?? "Unknown Error")")
+        return completion(nil, .internalServerError)
     }
-  }</code></pre>
+    // Iterate through result set here
+}</code></pre>
       <p>Here we check to see if the `selectResult` value is a valid Result set otherwise we log and error and return.</p>
```

Comment:
I think this should be worded the other way round... something like 
"The query should return a set of results, in the code we use `.asResultSet` to check that the returned value is a valid result set, otherwise we log an error and return."